### DISCUSSION TITLE
How to citation code

### DISCUSSION THREAD

**Discussion by KitHubb** [cafferychen777/ggpicrust2#77] [Q&A] [answered]
hello! Caffery Yang.
I'm enjoying your package. 

I would like to draw a plot of another statistical analysis (Wilcoxon rank sum test) using some code for the visualization output of ggpicrust2. 
When borrowing only part of the code, how should I cite it?

Your prompt response would be greatly appreciated.
thank you.

**Comment by cafferychen777** [comment, marked as answer]
Hello @KitHubb,

Thank you for reaching out and I'm glad to hear that you are enjoying the package.

Regarding your query about citing a portion of the code for your statistical analysis, you can directly reference it as follows:

Chen Yang, Jiahao Mai, Xuan Cao, Aaron Burberry, Fabio Cominelli, Liangliang Zhang, ggpicrust2: an R package for PICRUSt2 predicted functional profile analysis and visualization, Bioinformatics, Volume 39, Issue 8, August 2023, btad470, [https://doi.org/10.1093/bioinformatics/btad470](https://doi.org/10.1093/bioinformatics/btad470).

Additionally, I'm thrilled to hear that you've been working on your own package! When you're ready to publish it, I'd love if you could share it with me. I'm eager to see what you've created and how you've applied your skills and knowledge.

If you have any further questions or need additional assistance, feel free to ask.

Best regards,

Chen YANG

**Comment by KitHubb** [comment]
Hello. Thank you for your response.

thank you Since I only used part of the code, I wasn't sure whether to cite
GitHub or the paper.
However, I was able to achieve the result I wanted using your code. Thank
you very much

I have created a function that compares the relative abundance between
taxonomies using your code. Additionally, I have configured it to allow
sorting at the phylum level.

By adding several visual options, the code is still a bit complicated and
not optimized. However, I plan to modify it little by little.

The outcome created utilizing your code is as follows

[image: image.png]

It was very helpful.


Sincerely,

Soyeon Kim





2023년 11월 28일 (화) 오후 9:28, Caffery Yang ***@***.***>님이 작성:

> Hello @KitHubb <https://github.com/KitHubb>,
>
> Thank you for reaching out and I'm glad to hear that you are enjoying the
> package.
>
> Regarding your query about citing a portion of the code for your
> statistical analysis, you can directly reference it as follows:
>
> Chen Yang, Jiahao Mai, Xuan Cao, Aaron Burberry, Fabio Cominelli,
> Liangliang Zhang, ggpicrust2: an R package for PICRUSt2 predicted
> functional profile analysis and visualization, Bioinformatics, Volume 39,
> Issue 8, August 2023, btad470,
> https://doi.org/10.1093/bioinformatics/btad470.
>
> Additionally, I'm thrilled to hear that you've been working on your own
> package! When you're ready to publish it, I'd love if you could share it
> with me. I'm eager to see what you've created and how you've applied your
> skills and knowledge.
>
> If you have any further questions or need additional assistance, feel free
> to ask.
>
> Best regards,
>
> Chen YANG
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cafferychen777/ggpicrust2/discussions/77#discussioncomment-7691838>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZU7YO6SDMXNAKIA5UPAQRTYGXKINAVCNFSM6AAAAAA75VD64SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMOJRHAZTQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 

Kim soyeon, Graduate student

Jeonbuk National University Hospital Skin Microbiome Lab
T   063-259-3471    / M  010-5810-3103
E   ***@***.***
Clinical Research Support Center, Chonbuk National University Hospital, 20
Geonji-ro, Deokjin-gu, Jeonju-si, Jeollabuk-do, 54907
